prac_06/car.py

```diff
@@ -0,0 +1,36 @@
+"""CP1404/CP5632 Practical - Car class example."""
+
+
+class Car:
+    """Represent a Car object."""
+
+    def __init__(self, fuel=0, name=""):  # 6, Add a name field
+        """Initialise a Car instance.
+
+        fuel: float, one unit of fuel drives one kilometre
+        """
+        self.name = name  # 6, Add a name field
+        self.fuel = fuel
+        self._odometer = 0
+
+    def add_fuel(self, amount):
+        """Add amount to the car's fuel."""
+        self.fuel += amount
+
+    def drive(self, distance):
+        """Drive the car a given distance.
+
+        Drive given distance if car has enough fuel
+        or drive until fuel runs out return the distance actually driven.
+        """
+        if distance > self.fuel:
+            distance = self.fuel
+            self.fuel = 0
+        else:
+            self.fuel -= distance
+        self._odometer += distance
+        return distance
+
+    def __str__(self):  # 5 Add the __str__ method
+        """Return a string version of car object"""
+        return f"{self.name}, fuel={self.fuel}, odometer={self.odometer}"  # Task 6
```

Comment:
FYI: This file is missing a newline at the end, the hotkey <kbd>Alt</kbd> + <kbd>Shift</kbd> + <kbd>L</kbd> can be used in PyCharm to automatically format your code.